localstack/services/apigateway/integration.py

```diff
@@ -449,12 +449,14 @@ def invoke(self, invocation_context: ApiInvocationContext):
 
         # apply response templates
         response_content = json.dumps(remove_attributes(response, ["ResponseMetadata"]))
-        response_obj = requests_response(content=response_content)
+        invocation_context.response = response_obj = requests_response(content=response_content)
```

Comment:
Note: In one of the next refactoring iterations, we should unify the response handling for API GW integrations. Most of the `BackendIntegration` implementation are returning a `Response` from `invoke(..)` (even though it is not defined in the interface), and some of the implementations also update the `invocation_context.response` property on the context. 

We should think about how we can unify this (either having the response only on the context, or only returning it from `invoke`), to have a clear interface and make the request processing less ambiguous. 👍

Reply:
Agreed, if the integrations are self-contained as they're leaning to im not sure we will have the `invocation_context.response` anymore because all the work is done, meaning the response from the integration and the response templating is done. Food for thought, I guess